libs/sof/epipolar_curves.cpp

```
@@ -0,0 +1,165 @@
+constexpr float kAutoMinDepthLarge = 7.f;     // 7 m
+constexpr float kAutoMaxDepthSmall = 20.f;    // 20 m
+constexpr float kAutoMaxDepthLarge = 1000.f;  // 1 km
+
+// Fills any negative `min_depth` / `max_depth` with baseline-interpolated defaults, then throws
+// if the resulting range is not finite, strictly positive and monotonic. Only a negative value is
+// the "auto" trigger; 0 stays an explicit input and is rejected below rather than silently
+// replaced. The `!(x > 0)` / `!(x < y)` form rejects NaN; the isfinite checks add +inf, which
+// would otherwise reach `std::log` and turn every depth sample into NaN.
+void AutoDetectDepthRange(const float baseline, float& min_depth, float& max_depth) {
+  const float t = std::clamp((baseline - kAutoBaselineSmall) / (kAutoBaselineLarge - kAutoBaselineSmall), 0.f, 1.f);
+  if (min_depth < 0.f) {
```

Clang (14.0.6)

[warning] floating point literal has suffix 'f', which is not uppercase

(readability-uppercase-literal-suffix)